### PR TITLE
Fix hawk auth on search

### PR DESCRIPTION
**Pull request by richtier**

### CODE REVIEW

File: api/core/authentication.py [hunk 1]
```diff
@@ -36,7 +36,7 @@ def authenticate(self, request):
 
         from api.organisations.libraries.get_organisation import get_request_user_organisation_id
 
-        _, hawk_receiver = HawkOnlyAuthentication().authenticate(request)
+        hawk_receiver = _authenticate(request, _lookup_credentials)
```
Comment (richtier) [inline api/core/authentication.py:39]: doing the same thing but less complex

File: api/core/authentication.py [hunk 2]
```diff
@@ -125,13 +126,7 @@ def authenticate(self, request):
...
         """
 
-        try:
-            hawk_receiver = _authenticate(request, _lookup_credentials)
-        except HawkFail as e:
-            logging.error(f"Failed HAWK authentication {e}")
-            raise e
-
-        return AnonymousUser, hawk_receiver
+        return AnonymousUser, _authenticate(request, _lookup_credentials)
```
Comment (richtier) [inline api/core/authentication.py:129]: doing the same thing but less complex

File: api/search/application/views.py [hunk 3]
```diff
@@ -21,6 +22,7 @@ class MatchBoolPrefix(Query):
 class ApplicationDocumentView(DocumentViewSet):
     document = ApplicationDocumentType
     serializer_class = ApplicationDocumentSerializer
+    authentication_classes = (GovAuthentication,)
```
Comment (richtier) [inline api/search/application/views.py:25]: it turns out the frontend requires all endpoints to have hawk auth enabled

**Review by r4vi — APPROVED** [review]

File: api/search/application/views.py (hunk 3 above)
Comment (r4vi) [reply api/search/application/views.py:25]: that would explain it

**Pull request merged**